src/backends/all.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use crate::prelude::*;
+use anyhow::Result;
+
+macro_rules! append {
+    ($($backend:ident),*) => {
+        pub fn append(&mut self, other: &mut Self) {
+            $(
+                self.$backend.append(&mut other.$backend);
+            )*
+        }
+    };
+}
+macro_rules! is_empty {
+    ($($backend:ident),*) => {
+        pub fn is_empty(&self) ->bool {
+            $(
+                self.$backend.is_empty() &&
+            )* true
+        }
+    };
+}
+macro_rules! to_package_ids {
+    ($($backend:ident),*) => {
+        pub fn to_package_ids(&self) -> PackageIds {
+            PackageIds {
+                $(
+                    $backend: self.$backend.keys().cloned().collect(),
+                )*
+            }
+        }
+    };
+}
+
+macro_rules! x {
+    ($($backend:ident),*) => {
+        #[derive(Debug, Clone, Default)]
+        #[allow(non_snake_case)]
+        pub struct PackageIds {
+            $(
+                pub $backend: BTreeSet<<$backend as Backend>::PackageId>,
+            )*
+        }
+        impl PackageIds {
+            append!($($backend),*);
+            is_empty!($($backend),*);
+
+            pub fn difference(&self, other: &Self) -> Self {
+                Self {
+                    $(
+                        $backend: self.$backend.difference(&other.$backend).cloned().collect(),
+                    )*
+                }
+            }
+
+            pub fn to_install_options(self) -> InstallOptions {
+                InstallOptions {
+                    $(
+                        $backend: self.$backend.iter().map(|x| (x.clone(), <$backend as Backend>::InstallOptions::default())).collect(),
+                    )*
+                }
+            }
+            pub fn to_remove_options(self) -> RemoveOptions {
+                RemoveOptions {
+                    $(
+                        $backend: self.$backend.iter().map(|x| (x.clone(), <$backend as Backend>::RemoveOptions::default())).collect(),
+                    )*
+                }
+            }
+        }
+        impl core::fmt::Display for PackageIds {
```

Comment:
Why `core::fmt::Display`? You probably meant `std::fmt::Display` (Ik they are the same trait, but let's stick to `std` instead of `core`.